server/sonar-process-monitor/src/main/java/org/sonar/process/monitor/Monitor.java

```diff
@@ -282,10 +282,12 @@ private RestartorThread() {
 
     @Override
     public void run() {
+      stopProcesses();
       try {
-        stopProcesses();
         startProcesses();
       } catch (InterruptedException e) {
+        // Startup was interrupted. Processes are supposed to be stopped.
+        // Cleaning the interruption state
```

Comment:
`interrupt` is not cleaning the state, it's actually restoring it, ie. propagating it when applicable